src/main/java/com/yelp/nrtsearch/server/luceneserver/StartIndexHandler.java

```diff
@@ -58,9 +57,19 @@ public StartIndexResponse handle(IndexState indexState, StartIndexRequest startI
         final String primaryAddress;
         final int primaryPort;
         Path dataPath = null;
-        if (startIndexRequest.hasRestore()) {
-            RestoreIndex restoreIndex = startIndexRequest.getRestore();
-            dataPath = downloadArtifact(restoreIndex.getServiceName(), restoreIndex.getResourceName(), INDEXED_DATA_TYPE.DATA);
+        if (startIndexRequest.hasRestore() && !shardState.isStarted()) {
```

Comment:
This will make sure that we download the data and start index only once when restoring, but won't handle parallel start index requests without restore. Is that fine ?

Reply:
correct. Parallel start Index requests without restore are handled in [shardState.startPrimary](https://github.com/Yelp/nrtsearch/blob/master/src/main/java/com/yelp/nrtsearch/server/luceneserver/ShardState.java#L576) (or shardState.startReplica)  That method is synchronized and once a thread enters it, first thing it does is checks if index is already started and throws IllegalStateException